### ISSUE TITLE
Server Crashing Repeatedly (AdvancementDataPlayer)

### ISSUE THREAD

**Issue by Keribells** [PaperMC/Paper#3717]
### Link/paste of stack trace

https://paste.gg/p/anonymous/b3ad6304a8904351978857918c8eeaf0
https://paste.gg/p/anonymous/62394a78092941d789e253be013e0f37

___We need all of the stack trace! Do not cut off parts of it. Please do not use attachments. Upload your logs to a paste site or paste directly here.___

___Please do not make reports when the report says "DO NOT REPORT THIS TO PAPER". These are simply messages informing you of lag, to find what is causing it. Ask for tips on Discord or IRC instead.___

### Plugin list:
DiscordSRV
CoreProtect
GSit
Harbor
LuckPerms
Player Analytics
Vault
WorldEdit
WorldGuard

### Actions to reproduce (if known):
None known.

### Paper build number:
This server is running Paper version git-Paper-27 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

**Comment by Keribells** [comment]
May be similar issue to this one: #3413 

Attached playerdata for reference if it helps:
[playerdata.zip](https://github.com/PaperMC/Paper/files/4851069/playerdata.zip)

**Comment by hurrgurl** [comment]
Is there a way to completely disable advancements using paper?

Spigot doesn't seem to do it anymore.

**Comment by aikar** [comment]
@Keribells you didnt specify what UUID has the issue, since you gave entire folder....

**Comment by aikar** [comment]
Also note there is another folder called advancements that has the data in it, not player data files.

**Comment by aikar** [comment]
also need any data packs used.

**Comment by Keribells** [comment]
@aikar 

Not sure exactly which player is causing the issue as there was no one UUID logging in/out immediately before the crashes. But based on some process of elimination, if I had to make a guess, it would be 1e7bee9b-1600-46ca-8373-f824fb8c3a0b.

I've attached the advancements folder for reference now.

I'm not using any data packs on this server.
[advancements.zip](https://github.com/PaperMC/Paper/files/4857101/advancements.zip)
